lib/parallel.ml

```diff
@@ -0,0 +1,31 @@
+open Stdune
+
+let n_threads = 24
+
+let map ~f l =
+  let queue = ref l in
+  let queue_lock = Mutex.create () in
+  let rec worker result =
+    Mutex.lock queue_lock;
+    let hd =
+      match !queue with
+      | [] -> None
+      | hd :: tl ->
+          queue := tl;
+          Some hd
+    in
+    Mutex.unlock queue_lock;
```

Comment:
The protected queue code could be moved out of the function for readability. (eg. `seq`-style `protected_queue : 'a list -> (unit -> 'a option)`)